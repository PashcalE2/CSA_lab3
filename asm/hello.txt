org 0
v0_input: word v0_handler

org 0x20
msg: byte 'Hello, world!', 0
sent_char: byte 0

start:
    mov byte r2, 0
    ei
.wait:
    mov byte r1, [0x12]
    cmp byte r1, 0
    jne start.int
    jmp start.wait
.int:
    int 0
    cmp byte [msg+r2], 0
    jne start.wait
    halt

v0_handler:
    mov byte [0x12], 0
    mov byte [0x13], [msg+r2]
    inc r2
    iret