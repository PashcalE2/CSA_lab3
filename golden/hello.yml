in_source: |-
  org 0
  v0_input: word v0_handler

  org 0x20
  msg: byte 'Hello, world!', 0
  sent_char: byte 0

  start:
      mov byte r2, 0
  .wait:
      mov byte r1, [0x12]
      cmp byte r1, 0
      jne start.int
      jmp start.wait
  .int:
      int 0
      cmp byte [msg+r2], 0
      jne start.wait
      halt

  v0_handler:
      mov byte [0x12], 0
      mov byte [0x13], [msg+r2]
      inc r2
      iret

in_stdin: |-
  some
out_code: |-
  a
out_stdout: |
  a
out_log: |
  a
